src/Server/SSH/SSHSession.cpp

```diff
@@ -106,7 +101,13 @@ void SSHSession::handleKeyExchange()
 
 void SSHSession::disconnect()
 {
-    ssh_disconnect(session);
+    bool expected = false;
+    if (disconnected.compare_exchange_strong(expected, true))
+    {
+        static std::mutex global_disconnect_mutex;
```

Comment:
Needs a source code comment why we acquire a global mutex before disconnecting a SSH session (it is not intuitively clear).